File: algo_problems/q681-700/q682.java
class Solution {
    public int calPoints(String[] ops) {
        int i = 0;
        ArrayList<Integer> valueList = new ArrayList();
        while (i < ops.length) {
        	for (int v:valueList){
	        	System.out.print(v);
	        }
	        System.out.println();
    		int j = valueList.size();
    		int v = 0;
        	if (ops[i].equals("+")) {
        		if (j >= 2) {
        			v = valueList.get(j - 2) + valueList.get(j - 1);
        		}else if (j == 1){
        			v = valueList.get(0);
        		}
        		valueList.add(v);
        	}else if (ops[i].equals("D")) {
        		if (j != 0) {
        			v = valueList.get(j - 1) * 2;
        		}
        		valueList.add(v);
        	}else if (ops[i].equals("C")) {
        		valueList.remove(j - 1);
        	}else{
        		v = Integer.parseInt(ops[i]);
        		valueList.add(v);
        	}
        	i += 1;
        }
        int result = 0;
        for (int v:valueList){
        	result += v;
        }
        return result;
    }
}

File: algo_problems/q681-700/q682.py
class Solution:
    def calPoints(self, ops):
        """
        :type ops: List[str]
        :rtype: int
        """
        i = 0
        while i < len(ops):
            if ops[i] == '+':
                ops[i] = sum(ops[max(0, i - 2):i])
            elif ops[i] == 'D':
                if i == 0:
                    ops[i] = 0
                else:
                    ops[i] = 2 * ops[i - 1]
            elif ops[i] == 'C':
                ops.pop(i - 1)
                ops.pop(i - 1)
                i -= 1
                continue
            else:
                ops[i] = int(ops[i])
            i += 1
        return sum(ops)


if __name__ == '__main__':
    print(Solution().calPoints(["5", "2", "C", "D", "+"]))
